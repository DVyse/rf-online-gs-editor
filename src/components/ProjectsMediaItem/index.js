/**
 *
 * ProjectsMediaItem
 *
 */

import { Item, Icon } from 'semantic-ui-react';
import { Map } from 'immutable';
import { NavLink } from 'react-router-dom';
import moment from 'moment';
import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

function ProjectsMediaItem({ item, currentUser }) {
  const { title, description, name, id, createdAt, owner, isPublic } = item;
  const isCurrentIsOwner = !!(
    currentUser && currentUser.get('id') === owner.id
  );

  return (
    <Item>
      <Item.Content>
        <Item.Header as={NavLink} to={`/project/${id}`}>
          <Icon name={isPublic ? 'unlock' : 'lock'} />
          {title || name}
        </Item.Header>

        <Item.Meta>
          <small className="name">#{name}</small>
        </Item.Meta>
        <Item.Description>{description || 'â€”'}</Item.Description>
        <Item.Extra>
          <UserName active={isCurrentIsOwner}>
            <Icon name="user" />@{owner.login}
          </UserName>
          <Icon name="time" />
          {moment(createdAt).format('LLL')}
        </Item.Extra>
      </Item.Content>
    </Item>
  );
}

ProjectsMediaItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    createdAt: PropTypes.string.isRequired,
    isPublic: PropTypes.bool.isRequired,
    owner: PropTypes.shape({
      login: PropTypes.string.isRequired,
      id: PropTypes.string.isRequired,
    }),
  }).isRequired,
  currentUser: PropTypes.instanceOf(Map),
};

ProjectsMediaItem.defaultProps = {
  currentUser: null,
};

export default ProjectsMediaItem;

const UserName = styled.span`
  padding-right: 15px;
  ${({ active }) =>
    active &&
    `
    color: green;
    `};
`;
