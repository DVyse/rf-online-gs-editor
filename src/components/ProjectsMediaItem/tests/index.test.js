import React from 'react';
import { shallow } from 'enzyme';

import ProjectsMediaItem from '../index';

describe('<ProjectsMediaItem />', () => {
  it('Should exist', () => {
    const wrapper = shallow(
      <ProjectsMediaItem
        item={{
          id: '123',
          title: 'title',
          description: 'description',
          name: 'name',
          createdAt: '2018-01-01 06:00',
          isPublic: false,
          owner: {
            login: 'name',
            id: '456',
          },
        }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
