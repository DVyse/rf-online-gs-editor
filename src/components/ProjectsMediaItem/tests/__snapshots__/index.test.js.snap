// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectsMediaItem /> Should exist 1`] = `
<Item>
  <ItemContent>
    <ItemHeader
      as={[Function]}
      to="/project/123"
    >
      <Icon
        as="i"
        name="lock"
      />
      title
    </ItemHeader>
    <ItemMeta>
      <small
        className="name"
      >
        #
        name
      </small>
    </ItemMeta>
    <ItemDescription>
      description
    </ItemDescription>
    <ItemExtra>
      <ProjectsMediaItem__UserName
        active={false}
      >
        <Icon
          as="i"
          name="user"
        />
        @
        name
      </ProjectsMediaItem__UserName>
      <Icon
        as="i"
        name="time"
      />
      January 1, 2018 6:00 AM
    </ItemExtra>
  </ItemContent>
</Item>
`;
